File: backend/tests/fixtures/sample_vulnerable.js
// sample_vulnerable.js - Quantum Vulnerable Node.js Cryptography Sample

const crypto = require("crypto");
const elliptic = require("elliptic");

// 1. RSA Key Generation (Severity: High)
function generateRSA() {
  const { publicKey, privateKey } = crypto.generateKeyPairSync("rsa", {
    modulusLength: 2048,
  });
  return { publicKey, privateKey };
}

// 2. ECC Key Generation (Severity: High)
function generateECC() {
  const ec = new elliptic.ec("secp256k1");
  const key = ec.genKeyPair();
  const nodeECDH = crypto.createECDH("secp256k1");
  return { key, nodeECDH };
}

// 3. Diffie-Hellman Key Exchange (Severity: High)
function setupDH() {
  const aliceDH = crypto.createDiffieHellman(2048);
  aliceDH.generateKeys();
  return aliceDH;
}

// 4. Weak TLS Cipher Configuration (Severity: Medium)
const tlsOptions = {
  minVersion: "TLSv1.2",
  ciphers: "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256:TLS_RSA_WITH_AES_256_CBC_SHA256",
};

module.exports = { generateRSA, generateECC, setupDH, tlsOptions };

File: backend/tests/fixtures/sample_vulnerable.java
// sample_vulnerable.java - Quantum Vulnerable Java Cryptography Sample

import java.security.KeyPair;
import java.security.KeyPairGenerator;
import java.security.NoSuchAlgorithmException;
import java.security.interfaces.RSAPrivateKey;
import java.security.interfaces.ECPrivateKey;
import javax.crypto.KeyAgreement;

public class VulnerableCryptoService {

    // 1. RSA Key Pair Generation (Severity: High)
    public KeyPair generateRSAKeyPair() throws NoSuchAlgorithmException {
        KeyPairGenerator rsaGen = KeyPairGenerator.getInstance("RSA");
        rsaGen.initialize(2048);
        return rsaGen.generateKeyPair();
    }

    // 2. ECC / ECDSA Key Pair Generation (Severity: High)
    public KeyPair generateECCKeyPair() throws NoSuchAlgorithmException {
        KeyPairGenerator ecGen = KeyPairGenerator.getInstance("EC");
        return ecGen.generateKeyPair();
    }

    // 3. Diffie-Hellman Key Agreement (Severity: High)
    public KeyAgreement setupDiffieHellman() throws NoSuchAlgorithmException {
        KeyAgreement keyAgree = KeyAgreement.getInstance("DH");
        return keyAgree;
    }

    // 4. Weak TLS / Cipher Suites Configuration (Severity: Medium)
    public static final String[] WEAK_CIPHERS = new String[] {
        "TLS_RSA_WITH_AES_256_CBC_SHA256",
        "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"
    };
}
